routers/api/v2/users.go

```diff
@@ -101,9 +102,19 @@ func (r *apiV2Router) Register(ctx *gin.Context) {
 		return
 	}
 
-	// TODO: add email verification (https://github.com/unicsmcr/hs_auth/issues/87)
-
 	ctx.Status(http.StatusOK)
```

Comment:
Yeah so I was thinking about this and ended up going with a solution like this.

The problem we have here is that we don't have support for transactions, so when a user is created, there's no easy way to remove it in case we get an error in this handler and by the time we send out the email, the user is already created. So when we get an error from sending an email we have 2 options:
 1.  Remove the user and send a 4xx or 5xx response (but then we also have the risk that we get an error while removing the user).
 2. Ignore the error from the email service and return a 2xx.

I ended up going with option 2 because I didn't want to make the `Register` operation even more complex than it already is and also handling the email service error isn't really critical since we have the `ResendEmailVerification` operation. If the user doesn't receive the email, they can just request a new one.

If we had transactions, we could just cancel the transaction if the call to the email service fails. This would be the ideal solution for the future.

Reply:
Sounds like a good compromise to me, thanks for the explanation 🙂 

Does MongoDB support transactions in the Go API?

Reply:
Yes https://godoc.org/go.mongodb.org/mongo-driver/mongo#Session

The mongo clients might need some redesigning to support transactions though.